tests/integration/workspace_access/test_tacl.py

```diff
@@ -9,6 +9,46 @@
 logger = logging.getLogger(__name__)
 
 
+def test_permission_for_files_anonymous_func(sql_backend, inventory_schema, make_group):
+    group_a = make_group()
+    group_b = make_group()
+    group_c = make_group()
+    group_d = make_group()
+
+    sql_backend.execute(f"GRANT READ_METADATA ON ANY FILE TO `{group_a.display_name}`")
+    sql_backend.execute(f"GRANT SELECT ON ANONYMOUS FUNCTION TO `{group_c.display_name}`")
+
+    tables = StaticTablesCrawler(sql_backend, inventory_schema, [])
+    grants = GrantsCrawler(tables)
+
+    tacl_support = TableAclSupport(grants, sql_backend)
+
+    migration_state = MigrationState(
+        [
+            MigratedGroup.partial_info(group_a, group_b),
+            MigratedGroup.partial_info(group_c, group_d),
+        ]
+    )
+    for crawler_task in tacl_support.get_crawler_tasks():
+        permission = crawler_task()
+        apply_task = tacl_support.get_apply_task(permission, migration_state)
+        if not apply_task:
+            continue
+        apply_task()
+
+    any_file_actual = {}
+    for any_file_grant in grants._grants(any_file=True):
+        any_file_actual[any_file_grant.principal] = any_file_grant.action_type
+
+    assert group_b.display_name in any_file_actual
```

Comment:
Can you also assert the action type that new groups got as part of reassignment?

Reply:
Added the assertions